src/main/java/com/example/videotecha/model/Movie.java

```diff
@@ -0,0 +1,115 @@
+package com.example.videotecha.model;
+
+import com.example.videotecha.model.enums.Genre;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.SequenceGenerator;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+@Entity
+public class Movie {
+
+    @Id
+    @SequenceGenerator(name = "movieSequenceGenerator", sequenceName = "movieSequenceGenerator", initialValue = 1, allocationSize = 1)
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "movieSequenceGenerator")
+    @Column(name="id", unique=true, nullable=false)
+    private Long id;
+
+    @Column(nullable=false)
+    private String name;
+
+    @Column(nullable=false)
+    private String director;
+
+    @Enumerated(value = EnumType.STRING)
+    private List<Genre> genres = new ArrayList<>();
```

Comment:
Check this out: https://stackoverflow.com/questions/8969059/difference-between-onetomany-and-elementcollection